solutions/usaco-592.mdx

```diff
@@ -26,47 +25,103 @@ int getNum(int start, int direction) {
 	int radius = 1;
 	int prev = start;
 
-	while(prev >= 0 && prev < N) {
+	while (true) {
 		int next = prev;
-		while(next + direction >= 0 && next + direction < N 
-				&& abs(positions[next + direction] - positions[prev]) <= radius) {
+		// Getting the furthest explosion index without exceeding the current radius
+		while (next + direction >= 0 && next + direction < N 
+			&& abs(positions[next + direction] - positions[prev]) <= radius) {
 			next += direction;
 		}
 
-		if(next == prev) {
+		// We didn't find a new haybale, so the chain explosion is over
+		if (next == prev) {
 			break;
 		}
 
+		// Update our previous explosion and increment radius
 		prev = next;
 		radius++;
 	}
-
 	return abs(prev - start);
 }
 
-
 int main() {
 	cin.tie(0)->sync_with_stdio(0);
```

Comment:
not too good with c++, but would this be better?
```suggestion
	cin.tie(0) -> sync_with_stdio(0);
```

Reply:
no think the convention is no spaces for those arrows